proposals/memory-control/Overview.md

```diff
@@ -81,23 +124,69 @@ WebAssembly.instantiateStreaming(fetch('memory.wasm'), {js: { memory }})
+- Better toolchain integration: While LLVM supports multiple address spaces, most C/C++, Rust programs assume a single linear memory space, and that all pointers are accessible.
+- Graceful degradation for existing applications: This is a requirement for some larger applications, especially when running on older devices where updates aren't always possible - that there would be an easy way to polyfill this proposal. While this isn't a gating requirement, using a secondary memory does make this harder.
+
+Challenges of using a single linear memory:
+
+ - Significant API changes to decouple the current 1:1 mapping for a WebAssembly.Memory object to a JS Memory object.
+ - Potential performance degradation to default memory (however, the goal would be to minimize any effects to the default memory)
+ - Engine implementations are more involved.
+
+### Design decisions
+
+- **Aligning `addr` and `size` vs. requiring alignment and trapping:** Different operating systems have different rounding and alignment requirements for various memory APIs, so we cannot rely on the operating system to provide this behavior. This means that either the user or the WebAssembly runtime must align these parameters. We have opted to have the runtime align these parameters to avoid unnecessary trap conditions and codegen, and to hopefully make the APIs easier to use.
+
+- **Avoiding overlapping mappings:** TODO
+
 ### Alternative approaches
 
-#### Web API extensions
+#### "Out parameters" in Web APIs
 
 To support WebAssembly owning the memory and also achieving zero copy data transfer is to extend Web APIs to take typed array views as input parameters into which outputs are written. The advantage here is that the set of APIs that need this can be scaled incrementally with time, and it minimizes the changes to the WebAssembly spec.
 
 The disadvantages are that this would require changes to multiple Web APIs across different standards organizations, it’s not clear that the churn here would result in providing a better data transfer story as some APIs will still need to copy out.
 
-This is summarizing a discussion from the [previous issue](https://github.com/WebAssembly/design/issues/1162) in which this approach was discussed in more detail.
-
-#### Using GC Arrays
+This also only addresses the cost of copies, not the other motivators of this proposal.
 
-Though the proposal is still in phase 1, it is very probable that ArrayBuffers will be passed back and forth between JS/Wasm. Currently this proposal is not making assumptions about functionality that is not already available, and when available will evaluate what overhead it introduces with benchmarks. If at that time the mapping functionality is provided by the GC proposal without much overhead, and it makes sense to introduce a dependency on the GC proposal will be scoped to the other remaining functionality outlined above. 
```

Comment:
I struck this because the GC proposal is now in phase 4 without any features that I think would benefit this proposal.